File: client/client-main/src/app/service/subclass-diary/subclass-diary.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from 'src/environments/environment';

@Injectable({
  providedIn: 'root'
})
export class SubclassDiaryService {
  apiUrl=environment.apiUrl;

  constructor(private http: HttpClient) { }

  deleteSubclassDiary(data: any):Observable<number>{
    let params = new HttpParams()
      .set('stuId', data.stuId)
      .set('classId', data.classId);
    return this.http.delete<number>(this.apiUrl+"subclass-diary",{params})
  }
  
  updateSubclassDiary(data: any):Observable<number>{
    return this.http.put<number>(this.apiUrl+"subclass-diary",data);
  }
  findSubclassByStuId(stuId: any):Observable<any>{
    let params = new HttpParams()
      .set('stuId', stuId);
    return this.http.get<number>(this.apiUrl+"subclass-diary",{params})
  }

  findByStuIdAndClassId(data: any):Observable<any>{
    let params = new HttpParams()
      .set('stuId', data.stuId)
      .set('classId',data.classId);
    return this.http.get<number>(this.apiUrl+"subclass-diary/findByStuIdAndClassId",{params})
  }

  countStudentByClassId(classId: any):Observable<number>{
    let params = new HttpParams()
      .set('classId',classId);
      return this.http.get<number>(this.apiUrl+"subclass-diary/countStudentByClassId",{params})
  }
}

File: a2m_backend/a2m_backend-main/src/main/java/com/a2m/service/subclassdiary/SubclassDiaryService.java
package com.a2m.service.subclassdiary;

import java.util.List;

import com.a2m.domain.SubclassDiary;

public interface SubclassDiaryService {
	public int deleteSubclassDiary(Long stuId, Long classId);
	public int updateSubclassDiary(Long stuId, Long classId, String year,Boolean status);
	public List<SubclassDiary> findByStuId(Long stuId);
	public SubclassDiary findByStuIdAndClassId(Long stuId, Long classId);
	public int countStudentByClassId(Long classId);
}
